pbj-core/config/checkstyle/checkstyle.xml

```diff
@@ -0,0 +1,445 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?><!--
+    Palantir Baseline Checkstyle configuration.
+    Authors: Robert Fink, Brian Worth, Merrick Zoubeiri, and many other contributors. Based in part on http://checkstyle.sourceforge.net/google_style.html
+    Please keep checks alphabetized with one exception: "relaxed" checks are grouped together at the bottom for easier disabling.
+    Check-specific comments reference documents internal to Palantir and can be safely ignored or removed.
+ -->
+<!DOCTYPE module PUBLIC "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN" "https://checkstyle.org/dtds/configuration_1_3.dtd">
+<module name="Checker">
+    <property name="charset" value="UTF-8"/>
+    <property name="severity" value="error"/>
+
+    <module name="FileTabCharacter"/> <!-- Java Style Guide: Whitespace characters -->
+    <module name="NewlineAtEndOfFile"> <!-- Java Style Guide: Line ending: LF -->
+        <property name="lineSeparator" value="lf"/>
+    </module>
+    <module name="RegexpSingleline"> <!-- No reference needed as this is evident. -->
+        <property name="format" value="&lt;&lt;&lt;&lt;&lt;&lt;&lt;"/>
+        <property name="message" value="Found (&lt;&lt;&lt;&lt;&lt;&lt;&lt;), so it looks like you had a merge conflict that compiles. Please fix it."/>
+    </module>
+    <module name="RegexpSingleline"> <!-- No reference needed as this is evident. -->
+        <property name="format" value="&gt;&gt;&gt;&gt;&gt;&gt;&gt;"/>
+        <property name="message" value="Found (&gt;&gt;&gt;&gt;&gt;&gt;&gt;), so it looks like you had a merge conflict that compiles. Please fix it."/>
+    </module>
+    <module name="RegexpSingleline">
+        <property name="format" value="\s+$"/>
+        <property name="message" value="Whitespace at end-of-line"/>
+    </module>
+    <module name="RegexpMultiline"> <!-- Java Style Guide: Vertical Whitespace -->
+        <property name="fileExtensions" value="java"/>
+        <property name="format" value="^\n\n$"/>
+        <property name="message" value="Two consecutive blank lines are not permitted."/>
```

Comment:
Minor: this is a bit too strict. An extra blank line might be useful to visually separate blocks of code. When used wisely, they don't hurt.